src/MOIWrapper.jl

```diff
@@ -0,0 +1,283 @@
+    s = get_sense(instance.inner)
+    if s == :Max
+        return MOI.MaxSense
+    else
+        return MOI.MinSense
+    end
+end
+
+function LQOI.get_number_variables(instance::CplexOptimizer)  
+    num_var(instance.inner)
+end
+
+function LQOI.add_variables!(instance::CplexOptimizer, N::Int)
+    add_vars!(instance.inner, zeros(Float64, N), fill(-Inf, N), fill(Inf, N))
+end
+
+function LQOI.delete_variables!(instance::CplexOptimizer, 
+                                first_col::Int, last_col::Int)
+    del_cols!(instance.inner, Cint(first_col), Cint(last_col))
+end
+
+function LQOI.solve_mip_problem!(instance::CplexOptimizer)
+    instance.inner.has_int = true
+    LQOI.solve_linear_problem!(instance)
+end
+
+function LQOI.solve_linear_problem!(instance::CplexOptimizer)  
+    optimize!(instance.inner)
+end
+
+function LQOI.get_termination_status(instance::CplexOptimizer)
+    stat = get_status_code(instance.inner) 
+               
+    if stat == 1 # CPX_STAT_OPTIMAL
+        return MOI.Success
+    elseif stat == 3 # CPX_STAT_INFEASIBLE
+        return MOI.InfeasibleNoResult
+    elseif stat == 4 # CPX_STAT_INForUNBD
+        return MOI.InfeasibleOrUnbounded
+    elseif stat == 2 # CPX_STAT_UNBOUNDED
+        return MOI.UnboundedNoResult
+    elseif stat in [12, 21, 22, 36] # CPX_STAT_*ABORT*_OBJ_LIM 
+        return MOI.ObjectiveLimit
+    elseif stat in [10,34] # CPX_STAT_*ABORT_IT_LIM
+        return MOI.IterationLimit
+    elseif stat == 53 # CPX_STAT_CONFLICT_ABORT_NODE_LIM
+        return MOI.NodeLimit 
+    elseif stat in [11, 25, 33, 39] # CPX_STAT_*ABORT*TIME_LIM
+        return MOI.TimeLimit
+    elseif stat == 5 # CPX_STAT_OPTIMAL_INFEAS
+        return MOI.NumericalError
+    
+    # MIP STATUS
+    elseif stat in [101, 102] # CPXMIP_OPTIMAL, CPXMIP_OPTIMAL_TOL
+        return MOI.Success
+    elseif stat == 103 # CPXMIP_INFEASIBLE
+        return MOI.InfeasibleNoResult
+    elseif stat == 119 # CPXMIP_INForUNBD
+        return MOI.InfeasibleOrUnbounded
+    elseif stat == 118 # CPXMIP_UNBOUNDED
+        return MOI.UnboundedNoResult
+    elseif stat == [105, 106] # CPXMIP_NODE_LIM*
+        return MOI.NodeLimit 
+    elseif stat in [107, 108, 131, 132] # CPXMIP_*TIME_LIM*
+        return MOI.TimeLimit               
+    else        
+        return MOI.OtherError
+    end
+end
+
+function LQOI.get_primal_status(instance::CplexOptimizer)
+    stat = get_solution_info(instance.inner)[3]
+    if stat == 1
+        return MOI.FeasiblePoint
+    else
+        return MOI.UnknownResultStatus
+    end
+end
+
+function LQOI.get_dual_status(instance::CplexOptimizer)
+    if instance.inner.has_int 
+        return MOI.UnknownResultStatus
+    end
+    stat = get_solution_info(instance.inner)[4]
+    if stat == 1
+        return MOI.FeasiblePoint
+    else
+        return MOI.UnknownResultStatus
+    end
+end
+
+function LQOI.get_variable_primal_solution!(instance::CplexOptimizer, result)
+    fill_solution(instance.inner, result)
```

Comment:
Nit: `get_` instead of `fill_`?

Reply:
Since there are already get functions with the same name, the name should be different for inplace functions. following our discussion yesterday, I will use `get_` with `!`